### ISSUE TITLE
AuthorizationPermissionMismatch when downloading model with azure cli credentials

### ISSUE THREAD

**Issue by GK-Matias** [Azure/azure-sdk-for-python#37546]
- **Package Name**: azure-ai-ml
- **Package Version**: 1.19.0
- **Operating System**: Ubuntu
- **Python Version**: 3.11.6

**Describe the bug**
prior to version 1.19.0  it was possible to download a model when authenticated with the azure cli credentials but from 1.19.0 and onwards, the following error message appeares:

`azure.core.exceptions.HttpResponseError: Operation returned an invalid status 'This request is not authorized to perform this operation using this permission.'`
`ErrorCode:AuthorizationPermissionMismatch`

**To Reproduce**
1. az login
2. Run this code:
```
from azure.ai.ml import MLClient
from azure.identity import AzureCliCredential

token = AzureCliCredential()

ml_client = MLClient(token, "<id>",  "<rg_name>", "<mlw_name>")

ml_client.models.download(
    name="<model_name>", version="<model_version>", download_path="<path>"
)
```

**Expected behavior**
Expect it to be possible to download the model with the cli credentials.

**Additional context**
Using azure-identity version 1.18.0

**Comment by FilBr** [comment]
I'm having the same issue with 

Package Name: azure-ai-ml
Package Version: 1.22.0
Operating System: Ubuntu
Python Version: 3.10.13

any updates?

**Comment by giorgiococci** [comment]
I'm facing the same error, now both with Python SDK and with AZ CLI (`az ml model download`, version 2.31.0 ). Any update on this??

**Comment by GK-Matias** [comment]
@achauhan-scc are you planning to do anything about this?

**Comment by achauhan-scc** [comment]
sorry, I missed to update the progress, issue was fixed in > 1.22.4
<img width="419" alt="Image" src="https://github.com/user-attachments/assets/7b0166e1-5f1c-4a82-8d1d-8cdafb3a59b9" />

<img width="437" alt="Image" src="https://github.com/user-attachments/assets/6d566ae1-4c0e-4b91-b602-edc81563d289" />
